src/state/actions/scigateway.actions.tsx

```diff
@@ -211,6 +213,14 @@ export const configureSite = (): ThunkResult<Promise<void>> => {
         const loadingPlugins = loadMicroFrontends.init(settings.plugins);
         loadingPromises.push(loadingPlugins);
 
+        // load dark mode preference from local storage into store
+        // otherwise, intial state is false
+        const darkModeLocalStorage = localStorage.getItem('darkMode');
+        if (darkModeLocalStorage) {
+          const preference = darkModeLocalStorage === 'true' ? true : false;
+          dispatch(loadDarkModePreference(preference));
+        }
+
```

Comment:
This should probably either be moved to the top of the settings loading, or before even doing the `settings.json` request to prevent the light theme "flash" if a user has a preference for dark mode. Additionally, it might also be worth using [`prefers-color-scheme`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) if there is no local storage set: see https://blog.bitsrc.io/how-to-sync-your-react-app-with-the-system-color-scheme-78c0ad00074b for info on how to use this